de.fu_berlin.inf.dpp.server/src/de/fu_berlin/inf/dpp/server/session/JoinSessionRequestHandler.java

```diff
@@ -0,0 +1,64 @@
+package de.fu_berlin.inf.dpp.server.session;
+
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+import org.jivesoftware.smack.PacketListener;
+import org.jivesoftware.smack.packet.Packet;
+
+import de.fu_berlin.inf.dpp.communication.extensions.JoinSessionRequestExtension;
+import de.fu_berlin.inf.dpp.net.IReceiver;
+import de.fu_berlin.inf.dpp.net.xmpp.JID;
+import de.fu_berlin.inf.dpp.session.ISarosSessionManager;
+import de.fu_berlin.inf.dpp.util.NamedThreadFactory;
+
+/**
+ * Listens for and handles JoinSessionRequests allowing clients to join server sessions.
+ */
+public final class JoinSessionRequestHandler {
+    private final ISarosSessionManager sessionManager;
+
+    /**
+     * Invitation Requests are handled very fast and are almost immediately dispatched
+     * into another executor in the {@link NegotiationHandler}. More then one thread is currently
+     * not necessary and would likely just introduce more overhead.
+     * We just want asynchronous execution and a queue.
+     */
+    private final ThreadPoolExecutor executor = new ThreadPoolExecutor(1, 1, Long.MAX_VALUE,
+        TimeUnit.NANOSECONDS, new ArrayBlockingQueue<Runnable>(10),
+        new NamedThreadFactory("JoinSessionRequestHandler-"),
+        new ThreadPoolExecutor.DiscardPolicy());
+    
+
+    private final PacketListener joinSessionRequestListener = new PacketListener() {
+        @Override
+        public void processPacket(final Packet packet) {
+            executor.execute(new Runnable() {
```

Comment:
I guess that this should throw an Exception because the queue is bounded. https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/Executor.html#execute(java.lang.Runnable)

Can you verify that ? If not (i.e this method blocks) then this behavior is wanted. Otherwise please handle the exception (just logging a warning that we are busy right now and cannot accept anymore requests is ok)

Reply:
Took me some time to get around testing this. It does throw an exception. should be fixed now.